1brc.awk

```diff
@@ -3,10 +3,14 @@ BEGIN {
 }
 
 {
-	mins[$1] = $2 < mins[$1] ? $2 : mins[$1]
-	maxs[$1] = $2 > maxs[$1] ? $2 : maxs[$1]
+	c = ++counts[$1]
+	if (c==1) {  # new entry
+		mins[$1] = maxs[$1] = $2
+	} else {
+		mins[$1] = $2 < mins[$1] ? $2 : mins[$1]
+		maxs[$1] = $2 > maxs[$1] ? $2 : maxs[$1]
+	}
```

Comment:
Thanks. Testing for zero is slightly more idiomatic AWK and slightly faster, so let's use post-increment and reverse the order of the if/else:

```suggestion
	if (counts[$1]++) {
		mins[$1] = $2 < mins[$1] ? $2 : mins[$1]
		maxs[$1] = $2 > maxs[$1] ? $2 : maxs[$1]
	} else {
		mins[$1] = maxs[$1] = $2  # new entry
	}
```